searx/engines/mullvad_leta.py

```diff
@@ -0,0 +1,200 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+# lint: pylint
+
+"""This is the implementation of the Mullvad-Leta meta-search engine
+
+This engine _REQUIRES_ that searxng operate within a Mullvad VPN
+
+If using docker, consider using gluetun for easily connecting to the Mullvad
+
+- https://github.com/qdm12/gluetun
+
+Otherwise, follow instructions provided by Mullvad for enabling the VPN on Linux
+
+- https://mullvad.net/en/help/install-mullvad-app-linux
+
+"""
+
+from typing import TYPE_CHECKING, List
+from httpx import Response
+from lxml import html
+from searx.enginelib.traits import EngineTraits
+from searx.locales import region_tag, get_official_locales
+from searx.utils import eval_xpath, extract_text, eval_xpath_list
+from searx.exceptions import SearxEngineResponseException
+
+if TYPE_CHECKING:
+    import logging
+
+    logger = logging.getLogger()
+
+traits: EngineTraits
+
+use_cache: bool  # non-cache use only has 100 searches per day!
+
+search_url = "https://leta.mullvad.net"
+
+# about
+about = {
+    "website": search_url,
+    "wikidata_id": 'Q47008412',  # the Mullvad id - not leta, but related
+    "official_api_documentation": 'https://leta.mullvad.net/faq',
+    "use_official_api": False,
+    "require_api_key": False,
+    "results": 'HTML',
+}
+
+# engine dependent config
+categories = ['general', 'web']
+paging = True
+max_page = 50
+time_range_support = True
+time_range_dict = {
+    "day": "d1",
+    "week": "w1",
+    "month": "m1",
+    "year": "y1",
+}
+
+
+def is_vpn_connected(dom: html.HtmlElement) -> bool:
+    """Returns true if the VPN is connected, False otherwise"""
+    connected_text = extract_text(eval_xpath(dom, '//main/div/p[1]'))
+    return connected_text != 'You are not connected to Mullvad VPN.'
+
+
+def assign_headers(headers: dict) -> dict:
+    """Assigns the headers to make a request to Mullvad Leta"""
+    headers['Accept'] = "text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,*/*;q=0.8"
+    headers['Accept-Encoding'] = "gzip, deflate, br"
+    # headers['Accept-Language'] =  "en-US,en;q=0.5"
+    headers['Content-Type'] = "application/x-www-form-urlencoded"
+    headers['DNT'] = "1"
+    headers['Host'] = "leta.mullvad.net"
+    headers['Origin'] = "https://leta.mullvad.net"
+    headers['Referer'] = "https://leta.mullvad.net/"
+    headers['Sec-Fetch-Dest'] = "document"
+    headers['Sec-Fetch-Mode'] = "navigate"
+    headers['Sec-Fetch-Site'] = "same-origin"
+    headers['Sec-Fetch-User'] = "?1"
+    headers['Sec-GPC'] = "1"
+    headers['TE'] = "trailers"
+    headers['Upgrade-Insecure-Requests'] = "1"
+    return headers
+
+
+def request(query: str, params: dict):
+    """pre-request callback
+    params<dict>:
+      method  : POST/GET
+      headers : {}
+      data    : {} # if method == POST
+      url     : ''
+      category: 'search category'
+      pageno  : 1 # number of the requested page
```

Comment:
This doc here does not describe what the request function of mulvad_leta does

Reply:
If you not know a good documentation, you can remove the doc-string.

Reply:
I'm just following the format of [other engines I've seen](https://github.com/searxng/searxng/blob/3edaa26dae80b1256c9460d59754475b6d7293ea/searx/engines/duden.py#L34).

With this, I'll assume the engines with these docstrings are actually wrong and just remove my comment.